src/osd/modules/input/input_retro.cpp

```diff
@@ -915,22 +907,16 @@ class retro_joystick_device : public event_based_device<KeyPressEventArgs>
 	void poll() override
 	{
 		event_based_device::poll();
-
 	}
 
 	void reset() override
 	{
-		int i,j;
-
-		for(i = 0; i < 4; i++)
-			for(j=0;j<RETRO_MAX_BUTTONS;j++)
-				joystate[i].button[j] = 0;	
+		memset(&joystate, 0, sizeof(joystate));
```

Comment:
isn't it going to reset only user 0 gamepad?

Reply:
All joystate entries.
```
#include <stdio.h>

typedef struct joystate_t
{
   int button[6];
   int a1[2];
   int a2[2];
   int a3[2];
} Joystate;

Joystate joystate[4];
Joystate joystate[4];

int  main(int argc, char const *argv[]) {
	printf("size of joystate: %lu\n", sizeof(joystate));
	printf("size of Joystate: %lu\n", sizeof(Joystate));
	return 0;
}
```
```
$ ./a
size of joystate: 192
size of Joystate: 48
```